java-checks/src/test/files/checks/StringLiteralDuplicatedCheck.java

```diff
@@ -23,11 +23,14 @@ default void foo() {
     }
   }
 }
-
+class AllConstants {
+  private static final String Constant1 = "allConstant";
+  private static final String Constant2 = "allConstant";
+}
 class ConstantAlreadyDefined {
 
   private static final String A = "constant";
-  private static final String B = "constant"; // Noncompliant [[secondary=31,36]] {{Use already-defined constant 'A' instead of duplicating its value here.}}
+  private static final String B = "constant"; // Noncompliant [[secondary=34,39]] {{Use already-defined constant 'A' instead of duplicating its value here.}}
```

Comment:
Duplication should not be detected when literal is used to initialize another constant. This issue should only be raised on line 34.